### PR TITLE
#691 - adding setting per skill

### LINKED ISSUE #691: Add local settings/storage for skills
A mechanism for skills to store settings and data during and between runs. The data should be stored in the skill directory.

### PR DESCRIPTION
**Pull request by forslund**
This feature adds a `settings` property to the `MycroftSkill` class and automatically stores the data in the skill's directory when the skill shuts down (in a `settings.json` file). 
This resolves #691. 

Currently the settings are stored using the normal `json.dump()` mechanism which supports the most common datatypes. More complex datatypes can be added with manual serialization. An alternative could be to serialize the the internal dict using `pickle` but this is not considered safe.

### CODE REVIEW

File: mycroft/skills/settings.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# Mycroft Core is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Mycroft Core.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+    This module provides the SkillSettings dictionary, which is an simple
```
Comment (penrods) [inline mycroft/skills/settings.py]: Typo: "a simple", not "an"

Comment (forslund) [inline mycroft/skills/settings.py]: Fixed. (and rebased)

**Comment by augustnmonteiro** [comment]
Please rebase : D

**Comment by forslund** [comment]
Done!

**Pull request merged**